tests/Builder/Query/TypeOperatorTest.php

```diff
@@ -32,7 +33,7 @@ public function testQueryingByDataType(): void
                 zipCode: Query::type(2),
             ),
             Stage::match(
-                zipCode: Query::type('string'),
+                zipCode: QueryType::String,
```

Comment:
I can customize the `Query::type` factory to accept enum values and convert them to string if necessary.

Reply:
Not sure if that is necessary. This allows people to do whatever they want with the `$type` operator, but provides a nice shorthand for all known types.

Reply:
I assume this only works because QueryType is both an enum _and_ an OperatorInterface (and FieldQueryInterface in that it's associated with a field path). I assume this convenience only works for a single type check, though. Trying to match `zipCode` with `[QueryType::String, ...]` in order to match multiple types would produce invalid syntax, correct?

And that's where `Query::type(...)` comes in.